pylops/basicoperators/directionalderivative.py

```diff
@@ -71,25 +59,41 @@ def FirstDirectionalDerivative(
     operator with :math:`\mathbf{v}` along the main diagonal.
 
     """
-    Gop = Gradient(dims, sampling=sampling, edge=edge, kind=kind, dtype=dtype)
-    if v.ndim == 1:
-        Dop = Diagonal(v, dims=[len(dims)] + list(dims), axis=0, dtype=dtype)
-    else:
-        Dop = Diagonal(v.ravel(), dtype=dtype)
-    Sop = Sum(dims=[len(dims)] + list(dims), axis=0, dtype=dtype)
-    ddop = LinearOperator(Sop * Dop * Gop)
-    ddop.dims = ddop.dimsd = dims
-    ddop.sampling = sampling
-    return ddop
-
-
-def SecondDirectionalDerivative(
-    dims: InputDimsLike,
-    v: NDArray,
-    sampling: int = 1,
-    edge: bool = False,
-    dtype: DTypeLike = "float64",
-) -> LinearOperator:
+
+    def __init__(self, dims: InputDimsLike,
+                 v: NDArray,
+                 sampling: int = 1,
+                 edge: bool = False,
+                 kind: str = "centered",
+                 dtype: DTypeLike = "float64",
+                 name: str = 'F'):
+        self.dims = dims
+        self.sampling = sampling
+        self.edge = edge
+        self.kind = kind
+        self.dtype = dtype
```

Comment:
`dims` and `dtype` are somewhat "protected" and should probably not be defined manually. Most of the operators get them assign them via `super().__init__` call. In this case, the provided `self.Sop` will assign `dims` (see https://github.com/PyLops/pylops/blob/dev/pylops/linearoperator.py#L105).

My 2 cents for this and other similarly-related methods:
1. Make `_calc_first_ddop` a staticmethod, passing each argument (dims, dtype, edge, v, etc.) manually
2. Do not store `dims`, `dtype` and other LinearOperator properties directly in `__init__`.
3. Do not store `Sop` in `self`. When calling super().__init__, it will already be stored as `self.Op`

Reply:
Done in classes `Laplacian`, `FirstDirectionalDerivative`, and `SecondDirectionalDerivative`.